programs/stake/src/rewards.rs

```diff
@@ -610,6 +625,62 @@ mod tests {
         );
     }
 
+    #[test]
+    #[should_panic]
+    fn test_overflow_on_umul_calculate_rewards() {
```

Comment:
Rather than creating two separate tests, how about parametrizing the `stake` passed to `new_stake` and using the `test_case` crate to combine the two tests?

Reply:
Sorry, missed these comments. 

I'll see if i can write a test case that panics in multiple places.

Reply:
Neat! Updated to used test cases.